docs/articles/documentation/test-api/selecting-page-elements/selectors.md

```diff
@@ -138,45 +142,149 @@ You can initialize a selector with any of these objects.
     });
     ```
 
-### Initializers that Return Multiple Nodes
+### Check Selector Matching Set Length
+
+Functions and CSS selector strings which initialize a selector can have one matching DOM element on the page, several or nothing.
+Selector provides the following properties to check this:
+
+Property | Type | Description
+------ | ----- | -----
+`exists` | Boolean | `true` if at least one matching element exists.
+`count` | Number | The number of matching elements.
+
+ 
+```js
+import { Selector } from 'testcafe';
+import { expect } from 'chai';
+
+fixture `Example page`
+    .page `http://devexpress.github.io/testcafe/example/`;
 
-Functions and CSS selector stings that initialize a selector can return multiple DOM nodes.
-In this instance, you must filter these nodes to select a single node that will eventually be returned by the selector.
-Use the [text](selector-options.md#optionstext) and [index](selector-options.md#optionsindex) options for this.
+test('My test', async t => {
+    const osCount            = await Selector('.column.col-2 label').count;
+    const submitButtonExists = await Selector('#submit-button').exists;
+
+    expect(osCount).eql(3);
+    expect(submitButtonExists).is.true;
+});
+
+```
 
-The [text](selector-options.md#optionstext) option specifies text content of the node that should be returned
-or a regular expression that matches this content.
+Note that selector property getters are asynchronous.
 
-The [index](selector-options.md#optionsindex) option specifies the index of the node that should be returned.
+## Combined Selectors
 
-The following example demonstrates how to use the `text` and `index` options.
+Selector api provides methods that can be combined together, making flexible functional-style selector system.
```

Comment:
The selector API provides methods that can be combined together, thus providing you with a flexible functional-style selector mechanism.